pydantic/_internal/_decorators.py

```diff
@@ -322,16 +324,28 @@ def gather_decorator_functions(cls: type[Any]) -> DecoratorInfos:
             shimmed_func = var_value.shim(func) if var_value.shim is not None else func
             info = var_value.decorator_info
             if isinstance(info, ValidatorDecoratorInfo):
-                res.validator[var_name] = Decorator(var_name, shimmed_func, func, info)
+                res.validator[var_name] = Decorator(cls.__name__, var_name, shimmed_func, func, info)
             elif isinstance(info, FieldValidatorDecoratorInfo):
-                res.field_validator[var_name] = Decorator(var_name, shimmed_func, func, info)
+                res.field_validator[var_name] = Decorator(cls.__name__, var_name, shimmed_func, func, info)
             elif isinstance(info, RootValidatorDecoratorInfo):
-                res.root_validator[var_name] = Decorator(var_name, shimmed_func, func, info)
+                res.root_validator[var_name] = Decorator(cls.__name__, var_name, shimmed_func, func, info)
             elif isinstance(info, FieldSerializerDecoratorInfo):
-                res.field_serializer[var_name] = Decorator(var_name, shimmed_func, func, info)
+                # check whether a serializer function is already registered for fileds
+                for field_ser_dec in res.field_serializer.values():
+                    # check serializers function only in same model. serializer functions
+                    # for same field in subclass won't be cosiderd as duplicate
+                    if field_ser_dec.cls_name != cls.__name__:
```

Comment:
I would suggest using something like `pydantic._internal._core_utils.get_type_ref(cls)` instead of `cls.__name__` (and change `self.cls_name` to `self.cls_ref` or similar), since it's possible (and in some cases not unreasonable) to create subclasses with the same name, but `get_type_ref` will always produce a unique string for each unique class.

Reply:
You are right. I've changed it as you suggested